site/content/docs/howto/configure-auth-for-webapps.md

```diff
@@ -0,0 +1,347 @@
+        # not need to see the group memberships of the users.
+        - groups 
+    ```
+
+3. A web application which is allowed to use the Supervisor for authentication, but cannot see the username or
+   group memberships of the authenticated users, and cannot perform actions on Kubernetes clusters.
+
+    ```yaml
+      allowedGrantTypes:
+        - authorization_code
+        - refresh_token
+      allowedScopes:
+        - openid
+        - offline_access
+    ```
+
+## Create a client secret for the OIDCClient
+
+For each OIDCClient created by the Supervisor administrator, the administrator will also need to generate a client
+secret for the client. The client secrets are random strings auto-generated by the Supervisor upon request.
+The plaintext secret will only be returned once upon creation.
+
+```sh
+cat <<EOF | kubectl create -f -
+apiVersion: clientsecret.supervisor.pinniped.dev/v1alpha1
+kind: OIDCClientSecretRequest
+metadata:
+  name: client.oauth.pinniped.dev-my-webapp-client # the name of the OIDCClient
+  namespace: supervisor # the namespace of the OIDCClient
+spec:
+  generateNewSecret: true
+EOF
+```
+
+The server will respond with the newly generated client secret, e.g.:
+
+```
+NAMESPACE               NAME                                        SECRET     TOTAL
+pinniped-supervisor     client.oauth.pinniped.dev-my-webapp-client  abc123     1
+```
+
+Take care to make a note of the secret. After it has been returned by the create API, there is no other way to
+retrieve it in the future. The secret is not stored in plaintext on the server, which only stores a
+bcrypt-hashed version of the secret.
+
+This is the client secret that should be used, along with the client ID, by the web application when interacting
+with the Supervisor's OIDC token endpoint.
+
+## Rotating the client secret for an OIDCClient
+
+To facilitate rotating client secrets, an OIDCClient may have several active secrets. This enables the following process
+for the Supervisor administrator to change a client secret without causing web application downtime:
+
+1. Add a new, second secret to the OIDCClient by calling the create OIDCClientSecretRequest API again, as shown above.
+   Make note of the plaintext secret returned by the API. Now you have an old secret and a new secret, both of which will work.
+2. Reconfigure the web application to use the new client secret.
+3. Once the web application has been redeployed and is using the new client secret, call the client secret API again to
+   remove the old client secret:
+
+    ```sh
+    cat <<EOF | kubectl create -f -
+    apiVersion: oauth.virtual.supervisor.pinniped.dev/v1alpha1
+    kind: OIDCClientSecretRequest
+    metadata:
+      # the name of the OIDCClient
+      name: client.oauth.pinniped.dev-my-webapp-client
+      namespace: supervisor # the namespace of the OIDCClient
+    spec:
+      revokeOldSecrets: true
+    EOF
+    ```
```

Comment:
- There is a limit of 5 client secrets
- The newest client secret will have the least performance penalty 
- Older secrets still in use will accumulate performance penalties. IE, if the 3rd oldest secret is still in circulation, the BCrypt algorithm will be run against the newer 2 secrets before testing the 3rd, every time.  This will be a noticeable performance hit to every user.

Reply:
Done!